src/devBuilder/devBuilder.ts

```diff
@@ -177,6 +178,12 @@ export default abstract class DevBuilder<
   }
 
   protected async writeManifestScriptFile(fileName: string): Promise<string> {
+    const publicDirName = path.basename(this.publicDir);
+    if (fileName.split(path.sep)[0] === publicDirName) {
+      await copy(fileName, `${this.outDir}/${fileName}`);
```

Comment:
Does this copy need to happen? Wouldn't all public files have already been copied by line 67?

Reply:
I believe it is necessary because their behaviors are different.
line 67: `public/foo.js` -> `dist/foo.js`
this line: `public/foo.js` -> `dist/public/foo.js`

Reply:
Ah, ok. Line 67 is correct. The contents of public are supposed to be copied to the root of the dist directory not a subdirectory [[1]](https://vitejs.dev/guide/assets.html#the-public-directory) so we wouldn't want this line.

Reply:
If we include `public/foo.js` in the content scripts section of the manifest, it will not be found.